acceptance/tests/resource/mount/defined.rb

```diff
@@ -0,0 +1,36 @@
+test_name "defined should create an entry in filesystem table"
+
+confine :except, :platform => ['windows']
+
+fstab = '/etc/fstab'
+name = "pl#{rand(999999).to_i}"
+
+agents.each do |agent|
+  teardown do
+    #(teardown) restore the fstab file
+    on(agent, "mv /tmp/fstab #{fstab}", :acceptable_exit_codes => (0..254))
+    #(teardown) umount disk image
+    on(agent, "umount /#{name}", :acceptable_exit_codes => (0..254))
+    #(teardown) delete disk image
+    on(agent, "rm /tmp/#{name}", :acceptable_exit_codes => (0..254))
+    #(teardown) delete mount point
+    on(agent, "rm -fr /#{name}", :acceptable_exit_codes => (0..254))
+  end
+
+  #------- SETUP -------#
+  step "(setup) backup fstab file"
+  on(agent, "cp #{fstab} /tmp/fstab", :acceptable_exit_codes => [0,1])
+
+  #------- TESTS -------#
+  step "create a mount with puppet (defined)"
+  args = ['ensure=defined',
+          "device='/tmp/#{name}'"
```

Comment:
@kurtwall  that sounds like a bug in the provider not the test.

@joshcooper shouldn't the provider supply sane defaults for the `fstype` based on the OS detail?